traversal/graph/TraversalVertex.java

```diff
@@ -128,6 +132,22 @@ public static abstract class Properties {
         @Override
         public abstract int hashCode();
 
+        public boolean isThing() {
+            return false;
+        }
+
+        public boolean isType() {
+            return false;
+        }
+
+        public Thing asThing() {
+            throw TypeDBException.of(ILLEGAL_CAST, className(this.getClass()), className(TraversalVertex.Properties.Thing.class));
+        }
+
+        public Type asType() {
+            throw TypeDBException.of(ILLEGAL_CAST, className(this.getClass()), className(TraversalVertex.Properties.Type.class));
+        }
+
```

Comment:
in order to re-use the `GraphProcedure.Builder` methods to build a procedure from either a structure or a planner, we need to be able to cast `Parameters` into either `Thing` or `Type` parameters